### PR TITLE
detect: adds flow.age keyword

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5536

Describe changes:
- detect: adds flow.age keyword

PS : should we validate that `p->flow->lastts.tv_sec - p->flow->startts.tv_sec` >= 0 ?

I should create a meta ticket to list all the fields logged as integers which should have their detection keyword

### CODE REVIEW

File: doc/userguide/rules/flow-keywords.rst [hunk 1]
```diff
@@ -285,3 +285,22 @@ Format
...
+
+The time can be matched exactly, or compared using the _op_ setting::
+
+ flow.age:3    # exactly 3
+ flow.age:<3   # smaller than 3 seconds
+ flow.age:>=2  # greater or equal than 2 seconds
+
+Signature example::
+
+ alert tcp any any -> any any (msg:"Flow longer than one hour"; flow.age:>3600; sid:1; rev:1;)
```
Comment (victorjulien) [inline doc/userguide/rules/flow-keywords.rst:306]: this will alert on every packet after the flow hits 1 hour age, right?

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | tlpr1_stats_chk | | |
| .flow.memuse | 562615488 | 617486208 | 109.75% | 

Pipeline 9099

File: src/detect-flow-age.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#include "detect-parse.h"
+
+static int DetectFlowAgeMatch(
+        DetectEngineThreadCtx *det_ctx, Packet *p, const Signature *s, const SigMatchCtx *ctx)
+{
+    uint32_t age;
+    if (p->flow == NULL) {
+        return 0;
+    }
+    age = p->flow->lastts.tv_sec - p->flow->startts.tv_sec;
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/detect-flow-age.c:33]: ok

File: doc/userguide/rules/flow-keywords.rst (hunk 1 above)
Comment (catenacyber) [reply doc/userguide/rules/flow-keywords.rst:306]: That is not a very good example. Adding some `flowbit` then...

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/7848

**Pull request closed without merging**